src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt

```diff
@@ -429,6 +441,49 @@ class PlantingSiteStore(
     }
   }
 
+  fun fetchNonNotifiedSitesToScheduleObservations(): List<PlantingSiteId> {
+    requirePermissions { manageNotifications() }
+
+    return dslContext
+        .select(PLANTING_SITES.ID)
+        .from(PLANTING_SITES)
+        .where(PLANTING_SITES.SCHEDULE_OBSERVATION_NOTIFICATION_SENT_TIME.isNull)
+        .fetch { row -> row.value1() }
+        .filter { countReportedPlantsInSubzones(it).isNotEmpty() }
```

Comment:
This is going to have an N+1 problem with the filtering, though admittedly N will probably be small enough to not cause problems. Still, maybe better to do it in one query, something like
```kotlin
return dslContext
    .select(PLANTING_SITES.ID)
    .from(PLANTING_SITES)
    .where(PLANTING_SITES.SCHEDULE_OBSERVATION_NOTIFICATION_SENT_TIME.isNull)
    .andExists(
        DSL.selectOne()
            .from(PLANTINGS)
            .where(PLANTINGS.PLANTING_SITE_ID.eq(PLANTING_SITES.ID))
            .and(PLANTINGS.PLANTING_SUBZONE_ID.isNotNull))
    // This is a bit more efficient than the lambda
    .fetch(PLANTING_SITES.ID.asNonNullable())
```
and similarly for the query below this one.

Reply:
This is great 👍